crates/bevy_core_widgets/src/callback.rs

```diff
@@ -0,0 +1,46 @@
+use bevy_ecs::system::{Commands, SystemId, SystemInput};
+
+/// A callback defines how we want to be notified when a widget changes state. Unlike an event
+/// or observer, callbacks are intended for "point-to-point" communication that cuts across the
+/// hierarchy of entities. Callbacks can be created in advance of the entity they are attached
+/// to, and can be passed around as parameters.
+#[derive(Default, Debug)]
+pub enum Callback<I: SystemInput = ()> {
```

Comment:
Could we have a `&mut World` fn argument as an option? Or a boxed system design, and then move `System` -> `RegisteredSystem` and make it a secondary strategy?

Reply:
Not sure how your first suggestion would work with `Commands`.